clippy_lints/src/use_last.rs

```diff
@@ -0,0 +1,107 @@
+//! lint on using `x.get(x.len() - 1)` instead of `x.last()`
+
+use crate::utils::{match_type, paths, span_lint_and_sugg,
+                   snippet_with_applicability, SpanlessEq};
+use rustc::lint::{LateContext, LateLintPass, LintArray, LintPass};
+use rustc::{declare_tool_lint, lint_array};
+use rustc::hir::{Expr, ExprKind};
+use rustc_errors::Applicability;
+use syntax::ast::{LitKind};
+use if_chain::if_chain;
+
+/// **What it does:** Checks for using `x.get(x.len() - 1)` instead of `x.last()`.
+///
+/// **Why is this bad?** Using `x.last()` is easier to read and has the same result.
+///
+/// Note that using `x[x.len() - 1]` is semantically different from `x.last()`.
+/// Indexing into the array will panic on out-of-bounds accesses, while
+/// `x.get()` and `x.last()` will return `None`.
+///
+/// There is another lint (get_unwrap) that covers the case of using
+/// `x.get(index).unwrap()` instead of `x[index]`.
+///
+/// **Known problems:** None.
+///
+/// **Example:**
+///
+/// ```rust
+/// // Bad
+/// let x = vec![2, 3, 5];
+/// let last_element = x.get(x.len() - 1);
+///
+/// // Good
+/// let x = vec![2, 3, 5];
+/// let last_element = x.last();
+/// ```
+
+declare_clippy_lint! {
+    pub USE_LAST,
+    complexity,
+    "using `x.get(x.len() - 1)` instead of `x.last()`"
+}
+
+#[derive(Copy, Clone, Debug)]
+pub struct UseLast;
+
+impl LintPass for UseLast {
+    fn get_lints(&self) -> LintArray {
+        lint_array!(USE_LAST)
+    }
+
+    fn name(&self) -> &'static str {
+        "UseLast"
+    }
+}
```

Comment:
You should be able to use `declare_lint_pass!` here. [See](https://github.com/rust-lang/rust-clippy/issues/3917) and also [here](https://github.com/rust-lang/rust-clippy/pull/3968)